### ISSUE TITLE
six.moves throws E0401 import error

### ISSUE THREAD

**Issue by Sophie-Hawley** [PyCQA/pylint#1640]
### Steps to reproduce
from six.moves.urllib.parse import urlparse

### Current behavior
```
severity: 'Error'
message: 'E0401:Unable to import 'six.moves.urllib.parse''
at: '9,1'
source: 'pylint'
```
### Expected behavior
no errors

### pylint --version output
No config file found, using default configuration
pylint 1.7.2,
astroid 1.5.3
Python 3.6.2 (v3.6.2:5fd33b5, Jul  8 2017, 04:14:34) [MSC v.1900 32 bit (Intel)]

might be a regression of #223

**Comment by brycepg** [comment]
I can't reproduce this issue with:
```
pylint 1.7.2, 
astroid 1.5.3
Python 3.6.3 (default, Oct  9 2017, 12:11:29) 
[GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```
\> cat six_error.py:
```python
from six.moves.urllib.parse import urlparse
print(urlparse)
```

```
pylint six_error.py  -rn --enable=ALL

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

**Comment by brycepg** [comment]
For some reason

    from six.moves.urllib.parse import urlparse
    from six.moves.urllib import parse

raise a false positive while

    from six.moves import urllib

does not

**Comment by PCManticore** [comment]
You might want to check the implementation of our six brain plugin (https://github.com/PyCQA/astroid/blob/master/astroid/brain/brain_six.py), where `six`  magic is understood. I presume `urllib.parse` support is shaky at best, so it might require some improvements,.

**Comment by brycepg** [comment]
Thank you for the pointer @PCManticore

**Comment by megahall** [comment]
@PCManticore This change from @brycepg would help me out too, if you have some time to merge it at some point soon.
